check-redis/lib/check-redis.go

```diff
@@ -135,10 +136,59 @@ func checkReachable(args []string) *checkers.Checker {
 	)
 }
 
+type replicationOpts struct {
+	redisSetting
+	SkipMaster bool `long:"skip-master" description:"return ok if redis role is master"`
+}
+
+func checkReplication(args []string) *checkers.Checker {
+	opts := replicationOpts{}
+	psr := flags.NewParser(&opts, flags.Default)
+	psr.Usage = "replication [OPTIONS]"
+	_, err := psr.ParseArgs(args)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	c, info, err := connectRedisGetInfo(opts.redisSetting)
+	if err != nil {
+		return checkers.Unknown(err.Error())
+	}
+	defer c.Close()
+
+	if role, ok := (*info)["role"]; ok {
+		if role != "slave" && opts.SkipMaster {
```

Comment:
When role is "sentinel" and `--skip-master` is specified, this implementation reports as ok. but imo in that case it should return unknown?
https://redis.io/commands/role

Reply:
Oh sorry, `role` in `info` cannot be `sentinel`? If so, it's no problem.
https://redis.io/commands/INFO